packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -117,10 +121,16 @@ export class AnchorSet {
             }
             let dstPath: PathNode | undefined;
             if (toMove.length > 0) {
+                // There are anchors which are getting moved,
+                // therefor the destination needs to be created if it does not yet exist.
                 dstPath = this.trackInner(dst);
             } else {
+                // There are no anchors to move,
+                // therefor we want to avoid creating the destination if it does not already exist.
                 dstPath = this.find(dst);
                 if (dstPath !== undefined) {
+                    // Since we need a remove ref below to handle the `toMove.length > 0` case above,
+                    // add a ref here so that does not break this case.
                     dstPath.addRef();
```

Comment:
If we just deleted the else block, dstPath would be undefined and then we wouldn't go through the final block in the function. Is there anything we need to do there if there are no items to move?

Reply:
Yes: we need to update the anchors (if there are any) for everything in the dst after the insertion position (we need to increase their index by the number of items (not anchors) moved.